src/data-structures/linked-list/LinkedList.js

```diff
@@ -207,4 +207,88 @@ export default class LinkedList {
   toString(callback) {
     return this.toArray().map(node => node.toString(callback)).toString();
   }
+
+  /**
+   * Traverse through all nodes of the list from head to tail
+   * @param {*} callback
+   * @return {LinkedListNode[]}
+   */
+  traverse(callback = undefined) {
```

Comment:
It might be better to throw an error if the callback argument is not of type `function`.  `LinkedList` already has a `toArray` method, so it's never valid to pass a non-function arg.

If you verify that the callback is valid, you can simplify the method. Something like:
```js
traverse(callback) {
  if (typeof callback !== 'function') {
    throw new TypeError(
      `traverse method requires a callback function as an argument.\nArgument given: ${typeof callback}`
    )
  }

  let currentNode = this.head;
  const traversedNodes = [];

  while (currentNode) {
    traversedNodes.push(callback(currentNode.value));
    currentNode = currentNode.next;
  }

  return traversedNodes;
}
```